pkg/client/common.go

```diff
@@ -0,0 +1,82 @@
+package client
+
+import (
+	"fmt"
+
+	apistatus "github.com/nspcc-dev/neofs-api-go/pkg/client/status"
+	"github.com/nspcc-dev/neofs-api-go/v2/session"
+	"github.com/nspcc-dev/neofs-api-go/v2/signature"
+)
+
+// common interface of resulting structures with API status.
+type resCommon interface {
+	setStatus(apistatus.Status)
+}
+
+type statusRes struct {
```

Comment:
why we need this struct?

could be there any other fields in the future except status?

Reply:
It's shared between all `*Res` types in order to inherit `Status()` method. About other fields - possibly, not sure. I will write a comment explaining the purpose of the type.